harness/determined/core/_train.py

```diff
@@ -67,11 +68,12 @@ def _get_last_validation(self) -> Optional[int]:
         logger.debug(f"_get_last_validation() -> {steps_completed}")
         return steps_completed
 
-    def report_training_metrics(
+    def report_trial_metrics(
         self,
-        steps_completed: int,
-        metrics: Dict[str, Any],
-        batch_metrics: Optional[List[Dict[str, Any]]] = None,
+        metric_type: str,
+        total_batches: int,
+        metrics: Metrics,
```

Comment:
i'm not a fan of the separate `Metrics` class, makes it harder for users to tell the type.

Reply:
this is more subjective IMO:
some of the same benefits as https://github.com/determined-ai/determined/pull/7117#discussion_r1231462983
but also IDEs are good these days at showing what these are
![Screenshot 2023-06-15 at 12 45 50](https://github.com/determined-ai/determined/assets/12127420/ffcb2c30-dfbd-4108-9a0b-bc589d24c8d2)

I could see us wanting something more structured later or or refactor "metrics" which this would help

the type definition could also carry documentation with it in terms of comments, I think we do not allow absolutely anything that fits `Dict[str, Any]` do we?

Reply:
we allow anything that's json-serializable in metrics. 

yeah, it's pretty subjective. if you prefer the type aliases i won't stop you. my preference is still for explicit types when the object isn't a defined class. i can see how making it a class would be beneficial if later on we refactor it, but i'm also not sure that we _should_ refactor it to be anything more than a `Dict` type since adding complexity to `Metrics` would mean the user would have to construct it. 

looking at other more mature metrics APIs like ([WandB](https://docs.wandb.ai/ref/python/log) and [Ray](https://docs.ray.io/en/latest/ray-air/api/doc/ray.air.session.report.html), it seems like `Dict`/`Dict[str, Any]` is pretty common.

Reply:
> we allow anything that's json-serializable in metrics.

don't we disallow some non-numerical values in some cases? I recall getting bug reports about it.

While we are truly Dict[str, Any] it sounds good to expose it that way as well.